### PR TITLE
[libxml2] android and ios

### PR DESCRIPTION
**Pull request by m-kuhn**
**Add android and ios support to libxml2**

### CODE REVIEW

File: ports/libxml2/config.android.h [hunk 1]
```diff
@@ -0,0 +1,288 @@
+/* config.h.  Generated from config.h.in by configure.  */
```
Comment (ras0219) [inline ports/libxml2/config.android.h:1]: It would be nice to remove all the chaff from this file (comments, newlines) since it's being checked in.

**Review by ras0219 — APPROVED** [review]

**Comment by Neumann-A** [comment]
@ras0219 don't forget #17945 which removes the custom CMakeLists.txt

File: ports/libxml2/config.android.h (hunk 1 above)
Comment (m-kuhn) [reply ports/libxml2/config.android.h:1]: It's a generated file, I think there is value in leaving this (e.g. to compare it with files generated in other environments). But if it's desired I can remove it.

**Comment by m-kuhn** [comment]
> @ras0219 don't forget #17945 which removes the custom CMakeLists.txt

Feel free to close this one in favor of  #17945 anytime

**Comment by PhoebeHui** [comment]
Closing in favor of https://github.com/microsoft/vcpkg/pull/17945

**Pull request closed without merging**